src/query/service/src/interpreters/interpreter_unsetting.rs

```diff
@@ -46,19 +46,34 @@ impl Interpreter for UnSettingInterpreter {
         let mut keys: Vec<String> = vec![];
         let mut values: Vec<String> = vec![];
         let mut is_globals: Vec<bool> = vec![];
+        let settings = self.ctx.get_settings();
         for var in plan.vars {
             let (ok, value) = match var.to_lowercase().as_str() {
                 // To be compatible with some drivers
                 "sql_mode" | "autocommit" => (false, String::from("")),
                 setting => {
-                    let settings = self.ctx.get_settings();
-                    match settings.get_setting_level(setting)? {
-                        ScopeLevel::Global => {
-                            self.ctx.get_settings().try_drop_setting(setting)?;
-                        }
-                        ScopeLevel::Session => {}
+                    if matches!(settings.get_setting_level(setting)?, Global) {
+                        self.ctx.get_settings().try_drop_setting(setting)?;
                     }
-                    let default_val = settings.check_and_get_default_value(setting)?.to_string();
+                    let default_val = {
+                        if setting == "max_memory_usage" {
+                            let conf = self.ctx.get_config();
+                            if conf.query.max_memory_usage == 0 {
```

Comment:
No need to check `conf.query.max_memory_usage == 0`, just set the max_memory_usage to check_and_get_default_value?

Reply:
I see.